purchases/src/test/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsTrackerTest.kt

```diff
@@ -346,4 +349,63 @@ class DiagnosticsTrackerTest {
             })
         }
     }
+
+    // region Offline Entitlements
+
+    @Test
+    fun `trackEnteredOfflineEntitlementsMode tracks correct data`() {
+        val expectedProperties = mapOf<String, Any>()
+        every { diagnosticsFileHelper.appendEvent(any()) } just Runs
+        diagnosticsTracker.trackEnteredOfflineEntitlementsMode()
+        verify(exactly = 1) {
+            diagnosticsFileHelper.appendEvent(match { event ->
+                event.name == DiagnosticsEntryName.ENTERED_OFFLINE_ENTITLEMENTS_MODE &&
+                    event.properties == expectedProperties
+            })
+        }
+    }
+
+    @Test
+    fun `trackErrorEnteringOfflineEntitlementsMode tracks correct data for unknown error`() {
+        val expectedProperties = mapOf(
+            "offline_entitlement_error_reason" to "unknown",
+            "error_message" to "Unknown error.. Underlying error: test error message",
+        )
+        every { diagnosticsFileHelper.appendEvent(any()) } just Runs
+        diagnosticsTracker.trackErrorEnteringOfflineEntitlementsMode(
+            PurchasesError(
+                PurchasesErrorCode.UnknownError,
+                "test error message"
+            )
+        )
+        verify(exactly = 1) {
+            diagnosticsFileHelper.appendEvent(match { event ->
+                event.name == DiagnosticsEntryName.ERROR_ENTERING_OFFLINE_ENTITLEMENTS_MODE &&
+                    event.properties == expectedProperties
+            })
+        }
+    }
+
+    @Test
+    fun `trackErrorEnteringOfflineEntitlementsMode tracks correct data for one time purchase error`() {
+        val expectedProperties = mapOf(
+            "offline_entitlement_error_reason" to "one_time_purchase_found",
+            "error_message" to "There was a problem with the operation. Looks like we doesn't support that yet. Check the underlying error for more details.. Underlying error: Offline entitlements are not supported for active inapp purchases. Found active inapp purchases.",
```

Comment:
```suggestion
            "error_message" to "There was a problem with the operation. Looks like we don't support that yet. Check the underlying error for more details. Underlying error: Offline entitlements are not supported for active inapp purchases. Found active inapp purchases.",
```

Reply:
I'd also skip the "looks like we don't support that yet" bit entirely, I'm not sure we'll ever be able to support one-times in the first place. 
Also... I guess active inapp purchases is technically correct, but like, that sounds like any purchases, right? I'd maybe change the copy to `one time purchases` for clarity

Reply:
it'd be great to add a link to documentation too, we usually use the Rebrandly links for that kind of thing. It helps in case we ever want to document this very specifically, if we have a Rebrandly in place we can update where it points to later on

Reply:
> I'd also skip the "looks like we don't support that yet" bit entirely

That part is actually more generic, and it's the error message we display every time we get the `UnsupportedError` code... I could change it, but I would prefer to leave that part as is, since it might apply to something else... But happy to revamp this in our future better error handling exploration!